File: FrontEnd/src/app/models/qcm.ts
import { Question } from "./Question";


 
  export interface QCM {
    qcmId: number;
    titre: string;
    description: string;
    nombreQuestions: number;
    niveauId: number;
  }
  
  // models/qcmdto.ts
  export interface QCMDTO {
    QCMId :number;
    titre: string;
    description: string;
    questions: QuestionDTO[];
  }
  
  export interface QuestionDTO {
    texte: string;
    choix: ChoixDTO[];
  }
  
  export interface ChoixDTO {
    texte: string;
    estCorrect: boolean;
  }
  

File: BackEnd/AndexTech_Talent_Hub_Bibliothéque/Models/QCM.cs

using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;


namespace AndexTech_Talent_Hub_Bibliotheque.Models
{
    public class QCM
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int QCMId { get; set; }
        public string Titre { get; set; }
        public string Description { get; set; }
        public int NombreQuestions { get; set; }
        public int NiveauId { get; set; }


    }

}
